### PR TITLE
TRUNK-5939: Remove unused variable and correct JavaDoc links

### PR DESCRIPTION
**Pull request by sacull**
## Description of what I changed

I deleted two unused local variables, corrected four invalid JavaDoc links, and removed two outdated JavaDoc links.

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5939

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/api/db/hibernate/HibernatePersonDAO.java [hunk 1]
```diff
@@ -177,10 +177,8 @@ public void setSessionFactory(SessionFactory sessionFactory) {
 	}
 	
 	/**
-	 * @see org.openmrs.api.PersonService#getSimilarPeople(java.lang.String, java.lang.Integer,
-	 *      java.lang.String, java.lang.String)
-	 * @see org.openmrs.api.db.PersonDAO#getSimilarPeople(String name, Integer birthyear, String
-	 *      gender)
```
[the remark this answers is not in the document]
Comment (sacull) [reply api/src/main/java/org/openmrs/api/db/hibernate/HibernatePersonDAO.java:183]: @dkayiwa 
`PersonService#getSimilarPeople(java.lang.String, java.lang.Integer, java.lang.String, java.lang.String)` - does not exist in the PersonService class, that's why I corrected to `PersonService#getSimilarPeople(java.lang.String, java.lang.Integer, java.lang.String)`
Proof: https://github.com/openmrs/openmrs-core/blob/master/api/src/main/java/org/openmrs/api/PersonService.java#L97


`PersonDAO#getSimilarPeople(String name, Integer birthyear, String gender)` - it is not removed, only the word "gender" has been moved to the line above. This is by convention and improves readability.
Proof: https://github.com/openmrs/openmrs-core/pull/3525/files#diff-8418a7dc796909a947781080e9b24f83R181

**Comment by sacull** [comment]
@dkayiwa 
I can undo the JavaDoc fix, I did it by the way, but if it is a problem, just a word and I will restore the previous state. It's not a problem for me. ;)

**Comment by dkayiwa** [comment]
It would make the review easier if you stick to the ticket description. 😊

**Comment by sacull** [comment]
@dkayiwa Oh, I get it now. Remove JavaDoc patches from PR or leave it and promise that in the future I will only correct what is in the ticket description?

File: api/src/main/java/org/openmrs/api/db/hibernate/HibernatePersonDAO.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (dkayiwa) [reply api/src/main/java/org/openmrs/api/db/hibernate/HibernatePersonDAO.java:183]: Does javadoc require `java.lang.String` or `String name`?

[the remark this answers is not in the document]
Comment (sacull) [reply api/src/main/java/org/openmrs/api/db/hibernate/HibernatePersonDAO.java:183]: @dkayiwa There are two adjacent lines, two links, the first is `java.lang.String`, the second is `String name`. You can leave it, but does it not disturb the readability? Wouldn't it be better to replace it with 2x`java.lang.String` or 2x`String name`?

[the remark this answers is not in the document]
Comment (sacull) [reply api/src/main/java/org/openmrs/api/db/hibernate/HibernatePersonDAO.java:183]: https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#multiple@see

Oracle documentation accepts both forms, it is up to you to choose which form to use. ;)

**Comment by dkayiwa** [comment]
Ok. Merged for today. Hopefully you will keep your promise. 😊

**Comment by sacull** [comment]
@dkayiwa You will be proud of me yet, teacher. ;)

**Comment by dkayiwa** [comment]
Loool. You will be shocked to find yourself also teaching me some things here and there. :)

**Pull request merged**